### PR TITLE
[Monitor] az monitor metrics alert create: support RG and Sub as the scope values

### LINKED ISSUE #14350: [Monitor] az monitor metrics alert create: Support metric alerts in a resource group
RE: Azure Metric Alerts - Proposed CLI Commands. Item 12

### LINKED ISSUE #14351: [Monitor] az monitor metrics alert create: Support metric alerts in a subscription
RE: Azure Metric Alerts - Proposed CLI Commands. Item 13

### PR DESCRIPTION
**Pull request by MyronFanQiu**
Fix #14350
Fix #14351

### CODE REVIEW

**Comment by yonzhan** [comment]
Monitor

File: src/azure-cli/azure/cli/command_modules/monitor/operations/metric_alert.py [hunk 1]
```diff
@@ -213,17 +219,35 @@ def _parse_action_removals(actions):
...
+        nonlocal namespace
+        nonlocal resource_type
+        nonlocal scope_type
+        if namespace is None and resource_type is None and scope_type is None:
             namespace = item_namespace
             resource_type = item_resource_type
+            scope_type = item_scope_type
         else:
-            if namespace != item_namespace or resource_type != item_resource_type:
+            if namespace != item_namespace or resource_type != item_resource_type or scope_type != item_scope_type:
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/metric_alert.py:238]: Can we separate validation logic from main logic?

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/monitor/operations/metric_alert.py]: Done

File: src/azure-cli/azure/cli/command_modules/monitor/operations/metric_alert.py [hunk 2]
```diff
@@ -213,17 +218,41 @@ def _parse_action_removals(actions):
 
 
 def _parse_resource_type(scopes):
-    from msrestazure.tools import parse_resource_id
+    from azure.mgmt.core.tools import parse_resource_id
     from azure.cli.core import CLIError
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/monitor/operations/metric_alert.py:222]: Should't `from knack.util import CLIError` ?

Comment (jsntcy) [reply src/azure-cli/azure/cli/command_modules/monitor/operations/metric_alert.py]: It seems some codes use "from azure.cli.core.util import CLIError", so which one should we use?

**Pull request merged**